app/models/account_reset_request.rb

```diff
@@ -9,6 +11,24 @@ class AccountResetRequest < ApplicationRecord
              primary_key: 'issuer'
   # rubocop:enable Rails/InverseOf
 
+  # @return [AccountResetRequest, nil]
+  def self.pending_request_for(user)
+    where(
+      user: user,
+      granted_at: nil,
+      cancelled_at: nil,
+    ).where(
+      'requested_at > ?',
+      account_reset_wait_period_days(user).ago,
+    ).order(requested_at: :asc).first
+  end
+
+  def cancel!(now: Time.zone.now)
```

Comment:
One potential issue with this being a public method on the model is that someone could cancel any Account Reset Request. Whereas before, it was clear that it only applies to the ones that were pending.

Reply:
Maybe it could raise or short-circuit unless the request is pending?